src/s2client-fsharp/Sc2Game.fs

```diff
@@ -91,86 +104,115 @@ module Sc2Game =
                 let clients =
                     List.init clientPortsRequired (fun _ -> { Instance.ClientPort.BasePort = Instance.getFreePort();  Instance.ClientPort.GamePort = Instance.getFreePort() } )
 
-                { Instance.SharedPort = shared
-                  Instance.ServerPorts = server
-                  Instance.ClientPorts = clients }
+                { 
+                    Instance.SharedPort = shared
+                    Instance.ServerPorts = server
+                    Instance.ClientPorts = clients 
+                }
                 |> Some
-            else None                                  
+            else None
 
-        let playerIdTasks =
+        let getPlayerIds ports =
             participants
             |> List.map (fun part ->
+                let attachPart x = part, x
                 match part with
-                | Participant (instance, _, _)
-                | Observer (instance, _) ->
-                   Instance.joinGame instance gameSettings.UseFeatureLayers gameSettings.UseRender part.Simple ports
-                   |> Async.StartAsTask
-                   |> Some
-                | _ -> None)              
-        for playerIdTask in playerIdTasks do
-            match playerIdTask with
-            | Some t -> do! t |> Async.AwaitTask |> Async.Ignore
-            | None -> ()
-
-        let playerIds =
-            playerIdTasks |> List.map (Option.map (fun pit -> pit.Result))
-
-        return playerIds
+                |Participant (instance, _, _)
+                |Observer (instance, _) ->
+                    Instance.joinGame instance gameSettings.UseFeatureLayers gameSettings.UseRender part.Simple ports
+                    |> Async.RunSynchronously
+                    |> Result.map Some
+                    |> Result.map attachPart
+                |_ -> None |> attachPart |> Ok
+            )
```

Comment:
Note: Previous code was parallel while this is sequentially.

Reply:
I'll fix this.